### PR TITLE
[PRD-3638]-Check the platform credentials faster

### PR DESCRIPTION
**Pull request by EgorZhuk**
A new implementation + a flag to enable the old one + tests

### CODE REVIEW

**Comment by tmorgner** [comment]
Please use the existing Http-Client API instead of rolling your own Basic-Auth handler. The HTTP-Client will handle all sorts of authentication schemas for you, including the more secure digest schema, which every security conscious Administrator should enable by default. 

     Credentials defaultcreds = new UsernamePasswordCredentials("username", "password");
     client.getState().setCredentials(new AuthScope("myhost", 80, AuthScope.ANY_REALM), defaultcreds);

**Comment by EgorZhuk** [comment]
@tmorgner thank you, done.

**Comment by EgorZhuk** [comment]
By the way it will be easier to implement a story about "Ask for creds once" as far as we have a session cookie in the response.

File: designer/report-designer-extension-pentaho/src/main/java/org/pentaho/reporting/designer/extensions/pentaho/repository/actions/ValidateLoginTask.java [hunk 1]
```diff
@@ -91,4 +123,68 @@ public boolean validateLoginData() throws FileSystemException {
...
+      state.setCredentials( new AuthScope( uri.getHost(), uri.getPort(), AuthScope.ANY_REALM ), credentials );
+      httpClient.setState( state );
+
+      //Remove trailing slashes and spaces
+      final HttpMethod get = new GetMethod( uri.toString() );
+      final String authenticationString = loginData.getUsername() + ":" + loginData.getPassword();
+      //Checked with a real Pentaho server - it supports UTF encoding for a basic authentication
+      final String base64Credentials =
+        new Base64Encoder().encode( authenticationString.getBytes( Charset.forName( "UTF-8" ) ) );
+      get.setRequestHeader( AUTHORIZATION, "Basic " + base64Credentials );
```
Comment (tmorgner) [inline designer/report-designer-extension-pentaho/src/main/java/org/pentaho/reporting/designer/extensions/pentaho/repository/actions/ValidateLoginTask.java]: You dont need that manual base64Credentials stuff. The HTTP-Client's Credentials above take care of the authentication for you.

**Comment by EgorZhuk** [comment]
Oh, sorry, missed it somehow.

**Comment by EgorZhuk** [comment]
@tmorgner probably OK now

**Pull request merged**